pyk/src/pyk/cterm/show.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from ..kast.inner import KApply, KSort, KToken, flatten_label, top_down
+from ..kast.manip import free_vars, minimize_term
+from ..kast.prelude.k import DOTS
+from ..kast.pretty import PrettyPrinter
+from .cterm import CTerm
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable
+    from typing import Final
+
+    from ..kast.inner import KInner
+    from ..kast.outer import KDefinition, KFlatModule
+    from ..kast.pretty import SymbolTable
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+class CTermShow(PrettyPrinter):
+    minimize: bool
+    break_cell_collections: bool
+    omit_labels: tuple[str, ...]
+
+    def __init__(
+        self,
+        definition: KDefinition,
+        extra_unparsing_modules: Iterable[KFlatModule] = (),
+        patch_symbol_table: Callable[[SymbolTable], None] | None = None,
+        unalias: bool = True,
+        sort_collections: bool = True,
+        minimize: bool = True,
+        break_cell_collections: bool = True,
+        omit_labels: Iterable[str] = (),
+    ):
+        super().__init__(
+            definition,
+            extra_unparsing_modules,
+            patch_symbol_table,
+            unalias,
+            sort_collections,
+        )
+        self.minimize = minimize
+        self.break_cell_collections = break_cell_collections
+        self.omit_labels = tuple(omit_labels)
+
+    def let(
```

Comment:
If `PrettyPrinter` or the `Callable` is made an attribute, then this function can be removed altogether: the user can either set the remaining parameters (`minimize`, `break_cell_collections`, `omit_labels`) directly, or they can be made arguments of the `show*` methods.

Reply:
It would be nice for it to be possible to pass these arguments directyl to teh `show`. But I think overall, it ends up being less code to just set them once at class construction and go from there. Otherwise, all the points that are holding onto a `CTermShow` for printing need to also hold teh arguments `minimize` and others to determine how to print _that time_, instead of the user setting it up once up front.

Reply:
Makes sense.